packages/syft/src/syft/core/node/domain_client.py

```diff
@@ -759,7 +759,9 @@ def load_dataset(
             "\n\nRun `<your client variable>.datasets` to see your new dataset loaded into your machine!"
         )
 
-    def create_user(self, name: str, email: str, password: str, budget: int) -> dict:
+    def create_user(self, name: str, email: str, password: str, budget: float) -> dict:
+        if budget <= 0:
+            raise ValueError(f"Budget should be a positive number, but got {budget}")
         try:
             self.users.create(name=name, email=email, password=password, budget=budget)
```

Comment:
MAYBE: we could raise the exception in self.users.create (UserRequestsAPI)  function,
if they by  mistake call 
domain.users.create(**Data_Scientist_details)
Some of our old notebooks, create Data Scientist in this manner.

Reply:
Yep makes sense george 👍

Reply:
I still think we should have a check for the budget being negative in `def create` in `syft/core/node/common/client_manager/user_api.py`, as it is still the API being exposed to the client. Ideally, I think these kinds of checks would also lie on the server side in the user message service, but since we don't propagate those exceptions to the client currently, throwing those errors won't be communicated to the client anyway.

Reply:
From what I see it is something like:
- `create_user` is the API exposed to the user from the `client_side`.
- `create` is the method called by the `create_user` - the `user` should not directly call `create`, right?

I think we could do the followings:
* raise the exception in `create_user`
* raise the exception in `create`
* raise the exception in both places - to have a safeguard on the client-side and also server-side

If we raise the exception only in `create_user` the error is caught earlier in the calling chain and it does not forward the call to the server (by calling `create`).

The downside is that a `user` could change their part of the `create_user` implementation to remove that validation and in this case it could call `create` with a negative budget - is there any problem if there is a user created with a negative budget?

If we raise the exception in both places - we have a safeguard in both places until we figure out how to route exception from server to client side (now I have a preference for this method :smile:).
Also, if we choose to implement it also `server side` I would prefer to add a new parameter to the `create` method - `budget` (and also keep `**kwargs`). Is there any scenario when a user is created without explicitly specifying a budget?
(I prefer doing `if budget <= 0` raise error than `if "budget" in kwargs and kwargs["budget"] <= 0" raise error - if budget is always mandatory we could make it explicitly required -- but I am not sure if that is the case)

What do you think?

Reply:
Yeah, I like the idea of raising the exception in both places.
`is there any problem if there is a user created with a negative budget?` -> Ideally, nothing bad would happen but since the budget will always be >=0, so I feel it doesn't feel right to be able to assign a negative budget 😅 .
We can have `if budget < 0`, as we assign zero by default for new users when they register using `sy.register` method.